### PR TITLE
Add profiling flags (based on kubectl)

### PR DESCRIPTION
**Pull request by praveenrewar**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kapp/cmd/profiling.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		profile := pprof.Lookup(p.profileName)
+		if profile == nil {
+			return nil
+		}
+		f, err := os.Create(p.profileOutput)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		profile.WriteTo(f, 0)
```
Comment (prembhaskal) [inline pkg/kapp/cmd/profiling.go]: WriteTo can return error too. I think we should check it.

**Review by prembhaskal — COMMENTED** [review]
1 inline comment.

File: pkg/kapp/cmd/profiling.go (hunk 1 above)
Comment (praveenrewar) [reply pkg/kapp/cmd/profiling.go]: Done

**Comment by prembhaskal** [comment]
LGTM

**Review by 100mik — APPROVED** [review]
LGTM!

**Review by sethiyash — APPROVED** [review]
LGTM!

**Pull request merged**